### PR TITLE
Update `SCDataFrame` name to `CytoDataFrame`

### LINKED ISSUE #49: Rename `SCDataFrame` to `CytoDataFrame`
Add further clarity to the class by using the prefix "Cyto" which is more recognizable to the single-cell domain and aligns with other software packages surrounding that type of work.

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR updates the name of the class `SCDataFrame` to `CytoDataFrame` to help increase the alignment with other existing software in the same space and also to make the name more immediately recognizable to the single-cell domain.

A minor change to `poetry-dynamic-versioning` is also made to help avoid changes to the `__init__.py`. 

Closes #49 

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [ ] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: src/cosmicqc/__init__.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 """
 
 from .analyze import find_outliers
-from .frame import SCDataFrame
+from .frame import CytoDataFrame
 
 # note: version placeholder is updated during build
 # by poetry-dynamic-versioning.
-__version__ = "0.0.0"
+__version__ = "0.0.6.post3.dev0+1bfa792"
```
Comment (jenna-tomkinson) [inline src/cosmicqc/__init__.py:10]: I probably asked this question before but does this change automatically each PR?

**Review by jenna-tomkinson — APPROVED** [review]
Looks like every SCDataFrame was changed to the new name (I like it!) so I am approving this PR. Nice job and great small change!

File: src/cosmicqc/__init__.py (hunk 1 above)
Comment (d33bs) [reply src/cosmicqc/__init__.py]: Thanks for asking about this, it's a persisted version from a `poetry-dynamic-versioning` setting. I think this has been happening in error and I made an adjustment to the `pyproject.toml` in an effort to fix this as well.

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson for the review! Merging this in after making a change based on your comments.

**Pull request merged**